packages/cli/src/PublicApi/v1/handlers/workflows/workflows.service.ts

```diff
@@ -86,3 +89,29 @@ export async function updateWorkflow(workflowId: string, updateData: WorkflowEnt
 export function parseTagNames(tags: string): string[] {
 	return tags.split(',').map((tag) => tag.trim());
 }
+
+export async function getWorkflowTags(workflowId: string) {
+	return await Container.get(TagRepository).find({
+		select: ['id', 'name', 'createdAt', 'updatedAt'],
+		where: {
+			workflowMappings: {
+				...(workflowId && { workflowId }),
+			},
+		},
+	});
+}
+
+export async function updateTags(workflowId: string, newTags: string[]): Promise<any> {
+	await Db.transaction(async (transactionManager) => {
+		const oldTags = await Container.get(WorkflowTagMappingRepository).findBy({
+			workflowId,
+		});
+		if (oldTags.length > 0) {
+			await transactionManager.delete(WorkflowTagMapping, oldTags);
+		}
+		await transactionManager.insert(
+			WorkflowTagMapping,
+			newTags.map((tagId) => ({ tagId, workflowId })),
+		);
+	});
+}
```

Comment:
These could live in the tag service instead.

Reply:
This is a complicated bit, maybe we should use the `packages/cli/src/databases/repositories/workflowTagMapping.repository.ts` but that would also mean refactoring other parts of the code (such as internal API) and I think this falls outside the scope of this PR